File: src/test/kotlin/org/peyilo/booksource/TestBookSource.java
package org.peyilo.booksource;

import com.google.gson.Gson;
import org.junit.jupiter.api.Test;
import org.peyilo.booksource.entities.BookSource;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Arrays;

public class TestBookSource {

    @Test
    public void test2() throws IOException {
        Gson gson = new Gson();
        BookSource[] bookSources = gson.fromJson(new BufferedReader(new FileReader("F:\\Code\\IDEA\\BookSource\\src\\main\\resources\\b778fe6b.json")), BookSource[].class);
        System.out.println();
    }

    @Test
    public void test3() throws IOException {
        BookSource[] bookSources = ParseBookSourceKt.parseBookSource("F:\\Code\\IDEA\\BookSource\\src\\main\\resources\\shareBookSource.json");
        System.out.println(Arrays.toString(bookSources));
    }
}

File: src/test/kotlin/org/peyilo/booksource/TestBookSource.kt
package org.peyilo.booksource;

import com.google.gson.Gson
import org.junit.jupiter.api.Test
import org.peyilo.booksource.data.entities.BookSource
import org.peyilo.libreadview.loader.BiqugeBookLoader
import java.io.BufferedReader
import java.io.InputStreamReader

class TestBookSource {


    @Test
    fun testHello() {
        println("Hello TestBookSource")
    }

    @Test
    fun test1() {
        val gson = Gson()
        getResourceAsStream("b778fe6b.json").use { inputStream ->
            val reader = BufferedReader(InputStreamReader(inputStream))
            val bookSources: Array<BookSource> = gson.fromJson(reader, Array<BookSource>::class.java)
            println("book sources size: ${bookSources.size}")
        }
    }

    @Test
    fun test2() {
        val bookSource = getResourceAsStream("demo.json").use { inputStream ->
            parseBookSources(inputStream)!![0]
        }
        requireNotNull(bookSource) { "Book source not found" }
        println(bookSource.toString())

    }




}
